pylaia-htr-netout

```diff
@@ -11,16 +11,74 @@ import torch
 import torch.nn.functional as functional
 
 import laia.common.logging as log
+import laia.data.transforms as transforms
 from laia.common.arguments import add_argument, args, add_defaults
 from laia.common.loader import ModelLoader, CheckpointLoader
 from laia.data import ImageDataLoader, ImageFromListDataset
 from laia.engine.feeders import ImageFeeder, ItemFeeder
 from laia.experiments import Experiment
 from laia.losses.ctc_loss import transform_output
-from laia.utils import ImageToTensor
 from laia.utils.kaldi import ArchiveLatticeWriter, ArchiveMatrixWriter
 
 
+def output_matrix(
+    img_id,  # type: str
+    output,  # type: torch.Tensor
+    output_size,  # type: int
+    add_boundary_ctc_blank=False,  # type: bool
+    digits=10,  # type: int
+):
+    # type: (...) -> str
+    output = output.cpu()
+    matrix = "{} [\n".format(img_id)
+
+    def boundary_blank(cols):
+        # type: (int) -> str
+        matrix = "0 "
+        matrix += " ".join(["-1e+30"] * (cols - 1))
+        return matrix + "\n"
+
+    boundary = boundary_blank(output.size(1)) if add_boundary_ctc_blank else ""
+    matrix += boundary
+    matrix += (
+        "\n".join(
+            " ".join(
+                "{:.{digits}}".format(float(output[row, col]), digits=digits)
+                for col in range(output.size(1))
+            )
+            for row in range(output_size)
+        )
+        + "\n"
+    )
+    matrix += boundary
+    matrix += "]\n"
+    return matrix
+
+
+def output_lattice(
+    img_id,  # type: str
+    output,  # type: torch.Tensor
+    output_size,  # type: int
+    digits=10,  # type: int
+    **_  # type: **Any
+):
+    # type: (...) -> str
+    output = output.cpu()
+    matrix = "{}\n".format(img_id)
+    matrix += (
+        "\n".join(
+            "{:d}\t{:d}\t{:d}\t0,{:.{digits}},{:d}".format(
+                row, row + 1, col + 1, -float(output[row, col]), col + 1, digits=digits
+            )
+            for row in range(output_size)
+            for col in range(output.size(1))
+        )
+        + "\n"
+    )
+    matrix += "{:d}\t0,0,\n".format(output_size)
+    return matrix
+
+
```

Comment:
This should not be here, they were substituted by ArchiveMatrixWriter and ArchiveLatticeWriter.